### DISCUSSION TITLE
Uncaught TypeError: Cannot read properties of undefined (reading 'get')

### DISCUSSION THREAD

**Discussion by oliverbj** [wire-elements/modal#177] [Q&A]
Hi there.

I have a Laravel application, where I am trying to install Alpine.js (Using NPM) together with wire-elements/modal. In my app.js file I have this:

```javascript
require("./bootstrap");
require("alpinejs");

window.Alpine = Alpine;

Alpine.start();
```

I am loading the `app.js` file in my `app.blade.php` view file:

```
    <!-- Scripts -->
    <script src="{{ mix('js/app.js') }}" defer></script>
    <script src="{{ mix('js/modal.js') }}" ></script>
```

This gives me an error like the below though:

> Uncaught TypeError: Cannot read properties of undefined (reading 'get')

If I remove `Alpine.start();` from my app.js file, the error goes away (but Alpine is not working then).

**Comment by PhiloNL** [comment]
I would recommend adding the import before the Alpine definition:
```
require('../../vendor/wire-elements/modal/resources/js/modal')

window.Alpine = Alpine
Alpine.start()
```

**Reply by mikerockett** [reply]
Unfortunately, this doesn't work for me, using require or import.

```js
// js/app.js

import '../../vendor/wire-elements/modal/resources/js/modal'

// … font imports

import Alpine from 'alpinejs'
import mask from '@alpinejs/mask'

Alpine.plugin(mask)
Alpine.start()

window.Alpine = Alpine
```

```html
<!-- layouts/main.blade.php (stripped down) -->

<!DOCTYPE html>
<html>

<head>
  <meta charset="utf-8">
  <!-- ... -->
  <link rel="stylesheet" href="{{ mix('css/app.css') }}">
  @livewireStyles
  <script defer src="{{ mix('js/app.js') }}"></script>
  @stack('headScripts')
</head>

<body>
  {{ $slot }}
  @livewireScripts
  @livewire('livewire-ui-modal')
  @stack('bodyScripts')
</body>

</html>
```

Using Alpine 3.10.1, Livewire 2.10.5 and 1.0.6 of this package.

Am I doing something wrong, or has one of these releases broken something?

**Reply by mikerockett** [reply]
Oh wait, it actually works if I move the import to the end.

```js
// … font imports

import Alpine from 'alpinejs'
import mask from '@alpinejs/mask'

Alpine.plugin(mask)
window.Alpine = Alpine
Alpine.start()

import '../../vendor/wire-elements/modal/resources/js/modal'
```

Perhaps this should be documented?

**Comment by abishekrsrikaanth** [comment]
@PhiloNL How do you recommend we do this if we are using this package for laravel package development?

**Reply by PhiloNL** [reply]
I would recommend providing the same instructions to your end-users. Maybe part of the installation steps or FAQ.

**Reply by abishekrsrikaanth** [reply]
OK, but I am still getting this error when I try it on a new laravel project. Not sure what the issue seems to be.
